dialogue-core/src/main/java/com/palantir/dialogue/core/StickyAttachments.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.dialogue.Endpoint;
+import com.palantir.dialogue.Request;
+import com.palantir.dialogue.RequestAttachmentKey;
+import com.palantir.dialogue.Response;
+import com.palantir.dialogue.ResponseAttachmentKey;
+import com.palantir.dialogue.core.LimitedChannel.LimitEnforcement;
+import com.palantir.dialogue.futures.DialogueFutures;
+import java.util.Optional;
+import javax.annotation.CheckReturnValue;
+
+public final class StickyAttachments {
```

Comment:
Yep, I wasn't sure if we'd implement the finalized API pieces within dialogue-core or the client module. Given #1334 these should be package-private.

Reply:
Yeah, I strongly feel this should live here because that's how we can hide this complexity and never run into the problem you tried to handle here: https://github.com/palantir/dialogue/pull/1331#discussion_r670848625.